sw/device/tests/dif/dif_plic_sanitytest.c

```diff
@@ -117,131 +102,98 @@ static void uart_initialise(mmio_region_t base_addr, dif_uart_t *uart) {
   };
 
   // No debug output in case of UART initialisation failure.
-  if (dif_uart_init(base_addr, &config, uart) != kDifUartConfigOk) {
-    LOG_FATAL_AND_ABORT("UART init failed!");
-  }
+  CHECK(dif_uart_init(base_addr, &config, uart) == kDifUartConfigOk,
+        "UART init failed!");
 }
 
 static void plic_initialise(mmio_region_t base_addr, dif_plic_t *plic) {
-  if (dif_plic_init(base_addr, plic) != kDifPlicOk) {
-    LOG_FATAL_AND_ABORT("PLIC init failed!");
-  }
+  CHECK(dif_plic_init(base_addr, plic) == kDifPlicOk, "PLIC init failed!");
 }
 
 /**
  * Configures all the relevant interrupts in UART.
  */
-static bool uart_configure_irqs(dif_uart_t *uart) {
-  if (dif_uart_irq_enable(&uart0, kDifUartInterruptRxOverflow,
-                          kDifUartEnable) != kDifUartOk) {
-    LOG_ERROR("RX overflow IRQ enable failed!");
-    return false;
-  }
-  if (dif_uart_irq_enable(&uart0, kDifUartInterruptTxEmpty, kDifUartEnable) !=
-      kDifUartOk) {
-    LOG_ERROR("TX empty IRQ enable failed!");
-    return false;
-  }
+static void uart_configure_irqs(dif_uart_t *uart) {
+  CHECK(dif_uart_irq_enable(&uart0, kDifUartInterruptRxOverflow,
+                            kDifUartEnable) == kDifUartOk,
+        "RX overflow IRQ enable failed!");
+  CHECK(dif_uart_irq_enable(&uart0, kDifUartInterruptTxEmpty, kDifUartEnable) ==
+            kDifUartOk,
+        "TX empty IRQ enable failed!");
+}
 
-  return true;
+/**
+ * Cleans up all the relevant interrupts in UART. This ensures that further UART
+ * operations don't trip our interrupt handlers.
+ *
+ * TODO: Remove this one we add test configuration for doing "UART cleanup" in
+ * the test harness.
+ */
+static void uart_clean_up_irqs(dif_uart_t *uart) {
```

Comment:
Should be enough to simply reinitialise the UART. This reminds me of the other changes that we should start doing for DIFs (`init` split into `new`, 'configure' and 'reset').

In this test we also happen to clobber pre-test PLIC state, which is likely not an issues, but for robustness we probably should do some kind of arrangement similar to #2457 , but more general. I guess these is where your suggestion with a `typedef` in the header file comes in.

Reply:
> Should be enough to simply reinitialise the UART. 

Yes, this is what I meant.

Reply:
I have merged that PR, so I guess you could drop this function.